platform/ios/src/MGLMapView.mm

```diff
@@ -1128,9 +1128,68 @@ - (void)wakeGL:(__unused NSNotification *)notification
         [self validateLocationServices];
 
         [MGLMapboxEvents pushEvent:MGLEventTypeMapLoad withAttributes:@{}];
+        
+        [self updateLocalization];
+    }
+}
+
+- (void)updateLocalization
+{
+    NSString *mapboxLanguage = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLMapboxLanguage"];
+    NSString *language = [[[NSBundle mainBundle] preferredLocalizations] firstObject];
+    NSString *preferredLanguageCode = [self preferredLanguageCode];
+    
+    if (![mapboxLanguage isEqualToString:language] && ![mapboxLanguage isEqualToString:preferredLanguageCode]) {
+        mapboxLanguage = preferredLanguageCode;
+        [[NSUserDefaults standardUserDefaults] setObject:mapboxLanguage forKey:@"MGLMapboxLanguage"];
+        [self setLabelLanguage:mapboxLanguage];
+        [[NSNotificationCenter defaultCenter] postNotificationName:NSCurrentLocaleDidChangeNotification object:nil];
```

Comment:
I mentioned `NSCurrentLocaleDidChangeNotification` mainly because it’s possible to change the preferred languages without restarting on macOS. I figured this feature would go in the cross-platform code in MGLStyle rather than MGLMapView.